version: '3.8'

services:
  app-sade-sit:
    container_name: app-sade-sit
    restart: unless-stopped
    image: server-development-app-sade-sit
    environment:
      NODE_ENV: development
      PORT: 3001
      DB_HOST: db-sade-sit
      DB_USER: root
      DB_PASS: Root@123
      DB_NAME: stg_server1_sade
      JWT_SECRET: dummy-jwt-secret-key
      JWT_ACCESS_EXPIRATION_MINUTES: 720
      JWT_REFRESH_EXPIRATION_DAYS: 30
      LOG_FOLDER: logs/
      LOG_FILE: '%DATE%-app-log.log'
      LOG_LEVEL: error
      REDIS_HOST: localhost
      REDIS_PORT: 6379
      REDIS_USE_PASSWORD: no
      REDIS_PASSWORD: dummy-redis-password
    volumes:
      - ./../data/files:/app/files
      - ./../data/images:/app/images
    ports:
      - "3000:3001"
    networks:
      - sade-networks
    depends_on:
      - db-sade-sit

  myadmin-sade-sit:
    image: phpmyadmin/phpmyadmin
    container_name: myadmin-sade-sit
    restart: unless-stopped
    environment:
      PMA_HOST: db-sade-sit
    ports:
      - "1007:80"
    networks:
      - sade-networks

  db-sade-sit:
    image: mysql:5.7
    container_name: db-sade-sit
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: Root@123
      MYSQL_DATABASE: stg_server1_sade
    volumes:
      - db-data:/var/lib/mysql
    ports:
      - "3307:3306"
    networks:
      - sade-networks

networks:
  sade-networks:
    external: true

volumes:
  db-data:

